<p>This is not the solution for your problem. May be you can figure out what could have gone wrong with this.</p>

<p>Below is the method where "AssertionError" is thrown.</p>

<pre><code>/** Given a type, return its code for allocating arrays of that type.
*/
public static int arraycode(Type type)
{
    switch (type.tag)
    {
        case BYTE: return 8;
        case BOOLEAN: return 4;
        case SHORT: return 9;
        case CHAR: return 5;
        case INT: return 10;
        case LONG: return 11;
        case FLOAT: return 6;
        case DOUBLE: return 7;
        case CLASS: return 0;
        case ARRAY: return 1;
        default: throw new AssertionError("arraycode " + type);
    }
}
</code></pre>

<p>You can find the <strong>com.sun.tools.javac.jvm.Code</strong> class implementation <a href="http://www.javadocexamples.com/java_source/com/sun/tools/javac/jvm/Code.java.html" rel="nofollow">here</a> (I could not find the correct version of this class though, as in this class error is thrown at line 298 and in your case it is at 297). It might help.</p>

<p>I found link, <a href="http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=4901611" rel="nofollow">Bug ID = 4901611</a>, showing similar bug in previous release of Java. Version is Java 1.5. In this link you will find that this kind of error occurs when "generic method with vararg" gets unexpected input resulting in AssertionError. No workaround given and it was fixed in next update :(</p>
